examples/client/client.c

```diff
@@ -358,6 +360,70 @@ static void SetKeyShare(WOLFSSL* ssl, int onlyKeyShare, int useX25519,
         } while (ret == WC_PENDING_E);
     #endif
     }
+    #ifdef HAVE_LIBOQS
+    if (onlyKeyShare == 3) {
+        if (useLibOqs) {
+            int group = 0;
+
+            if (XSTRNCMP(oqsAlg, "KYBER512", XSTRLEN("KYBER512")) == 0) {
+                group = WOLFSSL_KYBER512;
```

Comment:
Is there a reason not to use the `groups[]` variable and let the `setGroups` via the `wolfSSL_set_groups` below?

Reply:
I didn't write that  part of the code.  But let me try to do what you suggest.

Reply:
Ah, I think I understand what is going on here.  (@kabuobeid please correct me if I'm wrong).

for all other "normal"  groups, we use wolfSSL_set_groups().  This saves startup time because it doesn't do keygen for all the groups you want to support.  You would only do keygen once you know which group was negotiated.  

For the liboqs groups, keygen can be potentially expensive so if we can pre-generate the public and private keys before connection then we would show a a potential performance gain during the connection.  So for liboqs groups, we call wolfSSL_UseKeyShare() which generates a keypair instead of adding the liboqs group to the groups list. as passing it into wolfSSL_set_groups(). 

This seems like a good approach to me.